src/include/add-on/add-on-workflow.rb

```diff
@@ -1504,11 +1491,8 @@ def RemoveProductWithDependencies
         return nil
       end
 
-      product_name = Ops.get_locale(
-        pi,
-        ["product", "display_name"],
-        Ops.get_locale(pi, ["product", "name"], _("Unknown product"))
-      )
+      product_name =  pi["product"].display_name ||
+        pi["product"].name || _("Unknown product")
```

Comment:
We can omit the default, at least the `.name` must be valid, the default is never used.